isotope-intensity-estimation.Rmd

```diff
@@ -309,26 +337,47 @@ save(cnf_frequency, cnf_per_cmpd, file = paste0(dr, "cnf_frequencies.RData"))
 Or do the same based on the mz differences with respect to monoisotopic mass
 ---->
 
-```{r, echo = FALSE, eval = file.exists(paste0("data/RData/isotope-intensity-estimation/"))}
+```{r, echo = FALSE, eval = file.exists(paste0(dr, "cnf_frequencies.RData"))}
 load(paste0(dr, "cnf_frequencies.RData"))
 ```
 
 
 ```{r}
-cnf_proportion <- sort(cnf_frequency / nrow(cmps), decreasing = TRUE)[-1]
+cnf_proportion <- sort(cnf_frequency / nrow(cmps), decreasing = TRUE)
 par(mar = c(8, 4.1, 4.1, 2.1))
 barplot(cnf_proportion[cnf_proportion > 0.5], las = 2)
 ```
 
-In the plot above we can see the families of configurations that for compounds 
-in HMBD produce a significant peak for at least a half of the compounds
-(the height of the bar specifies the exact proportion). 
-In other words, if we randomly sample a compund in HMBD the chance to observe a 
-significant peak related to the above families will be high. I observe however 
-that this approach penalizes families related to elements with low counts in 
-HMBD compounds. The number of times a family related to N, S or Cl is found to 
-be significant is limited by the fact that the number of compounds in HMBD that 
-contain these elements is much lower than those that contain C, H, or O.  
+In the plot above we can see isotopologues for compounds in HMBD that produce a
+*significant* peak for at least a half of the compounds (the height of the bar
+specifies the exact proportion). In other words, if we randomly sample a compund
+in HMBD the chance to observe a significant peak related to the above
+isotopologues will be high. I observe however that this approach penalizes
+families related to elements with low counts in HMBD compounds. The number of
+times a family related to N, S or Cl is found to be significant is limited by
+the fact that the number of compounds in HMBD that contain these elements is
+much lower than those that contain C, H, or O.
+
+We thus next compare the isotopologue frequency not against the total number of
+compounds in HMDB, but against the number of compounds in HMDB that contain each
+of the elements from the isotopologue definition in their chemical formula.
+
+```{r}
+ncomps <- vapply(names(cnf_frequency), function(x) {
+    x <- unlist(strsplit(x, split = "\\]"))
+    x <- gsub("[0-9\\[]+", "", x)
+    ## require at least one element for each element in the
+    ## isotopologue definition
+    sum(rowSums(counts_df[, x, drop = FALSE] > 0) == length(x))
+}, integer(1))
+```
+
+```{r, fig.height = 5, fig.width = 17}
```

Comment:
This is obviously different from your approach below - please have a look and then we should discuss (also with Vinicius) what's better.